### PR TITLE
Definition printer

### PR DESCRIPTION
**Pull request by crisu83**

### CODE REVIEW

File: src/GraphQL.php [hunk 1]
```diff
@@ -254,7 +254,7 @@ public static function execute(
      */
     public static function print(NodeInterface $node): string
     {
-        return static::make(PrinterInterface::class)->print($node);
+        return static::make(NodePrinterInterface::class)->print($node);
```
Comment (crisu83) [inline src/GraphQL.php]: Renamed this for consistency.

File: src/Language/NodePrinter.php [hunk 2]
```diff
@@ -366,7 +368,7 @@ protected function one(?NodeInterface $node): string
      * @param array $nodes
      * @return array
      */
-    protected function many(array $nodes): array
+    protected function printMany(array $nodes): array
```
Comment (crisu83) [inline src/Language/NodePrinter.php]: Renamed these methods based @hungneox 's suggestion.

File: src/Type/Definition/DefinitionInterface.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Digia\GraphQL\Type\Definition;
+
+/**
+ * Tagging interface for GraphQL definitions (schema, types, directive, etc.).
+ */
+interface DefinitionInterface
+{
+}
```
Comment (crisu83) [inline src/Type/Definition/DefinitionInterface.php]: Introduce this tagging interface to be able to type-hint classes within the `DefinitionPrinter`

File: src/Type/Definition/TypeInterface.php [hunk 4]
```diff
@@ -2,15 +2,10 @@
...
 {
     /**
      * @return string|null
      */
     public function getName(): ?string;
-
-    /**
-     * @return string|null
-     */
-    public function getDescription(): ?string;
```
Comment (crisu83) [inline src/Type/Definition/TypeInterface.php]: This was actually a bug, `ListType` and `NonNullType` cannot have a description.

File: src/Type/Definition/DescriptionAwareInterface.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Digia\GraphQL\Type\Definition;
+
+interface DescriptionAwareInterface
+{
+    /**
+     * @return null|string
+     */
+    public function getDescription(): ?string;
+}
```
Comment (crisu83) [inline src/Type/Definition/DescriptionAwareInterface.php]: Introduce this interface to be able to check whether or not the class has a description.

File: src/Schema/SchemaInterface.php [hunk 6]
```diff
@@ -9,6 +9,8 @@
 use Digia\GraphQL\Type\Definition\ObjectType;
 use Digia\GraphQL\Type\Definition\TypeInterface;
 
+// TODO: Consider removing this interface.
+
 interface SchemaInterface
```
Comment (crisu83) [inline src/Schema/SchemaInterface.php]: I think we should remove this interface. I don't see why you'd want to replace it. Besides, it's not registered with the container anyway. However, I think I'll do this in a separate PR.

File: src/Language/NodePrinter.php [hunk 7]
```diff
@@ -30,11 +32,11 @@
 use Digia\GraphQL\Language\Node\VariableNode;
 use function Digia\GraphQL\Util\toString;
 
-class Printer implements PrinterInterface
+class NodePrinter implements NodePrinterInterface
```
Comment (crisu83) [inline src/Language/NodePrinter.php]: After renaming this class I realized that we should probably move both this and the `NodeBuilder` into the `Node` namespace as it uses quite a few classes from that namespace and the class name also starts with the word `Node`.

Comment (fubhy) [reply src/Language/NodePrinter.php]: Agreed.

File: src/Schema/SchemaInterface.php (hunk 6 above)
Comment (fubhy) [reply src/Schema/SchemaInterface.php]: If it's not swappable anyways there is no point in having it indeed.

**Comment by crisu83** [comment]
I would actually like to merge this ASAP because it contains a lot of all-around improvements. I created a separate issue for the test cases (#211). @fubhy @hungneox are you ok with this?

**Pull request merged**